src/visualizers/widgets/LogViewer/AnsiParser.js

```diff
@@ -0,0 +1,246 @@
+                });
+                ++sIndex;
+            }
+        }
+
+        return result;
+    };
+
+    /**
+     * getAtIndex
+     * Returns the content and ANSI style at known index.
+     *
+     * @name getAtIndex
+     * @function
+     * @param {String} input The input string.
+     * @param {String} noAnsi The input string without containing ansi styles.
+     * @param {Number} index The character index.
+     * @return {Object} An object containing the following fields:
+     *
+     *  - `style` (String): The ANSI style at provided index.
+     *  - `content` (String): The content (character) at provided index.
+     */
+    AnsiParser.getAtIndex = function (input, noAnsi, index) {
+
+        if (typeof noAnsi === "number") {
+            index = noAnsi;
+            noAnsi = AnsiParser.removeAnsi(input);
+        }
+
+        if (input === noAnsi) {
+            return {
+                style: ""
+                , content: noAnsi[index]
+            };
+        }
+
+        var sIndex = 0
+            , eIndex = index
+            , color = []
+            , cmatch
+        ;
+
+        while (sIndex < input.length) {
+            cmatch = input.substr(sIndex).match(/^(\u001B\[[^m]*m)/);
+            if (cmatch) {
+                if (cmatch[0] == '\x1B[0m') {
+                    color = [];
+                } else {
+                    if (/^\u001b\[38;5/.test(cmatch[0]) && /^\u001b\[38;5/.test(color.slice(-1)[0])) {
+                        color.splice(-1);
+                    }
+                    color.push(cmatch[0]);
+                }
+                sIndex += cmatch[0].length;
+            } else {
+                if (!eIndex) {
+                    break;
+                }
+                ++sIndex;
+                --eIndex;
+            }
+        }
+
+        return {
+            style: color.join("")
+            , content: noAnsi[index]
+        };
+    };
+
+    /**
+     * removeAnsi
+     * Removes ANSI styles from the input string.
+     *
+     * @name removeAnsi
+     * @function
+     * @param {String} input The input string.
+     * @return {String} The string without ANSI styles.
+     */
+    AnsiParser.removeAnsi = function (input) {
+        return input.replace(/\u001b\[.*?m/g, "");
+    };
+
+    /**
+     * stringify
+     * Stringifies an array of objects in the format defined by `AnsiParser`.
+     *
+     * @name stringify
+     * @function
+     * @param {Array} arr The input array.
+     * @return {String} The stringified input array.
+     */
+    AnsiParser.stringify = function (arr) {
+
+        var str = ""
+            , cArr = null
+            , i = 0
+        ;
+
+        if (arr && Array.isArray(arr[0])) {
+            for (; i < arr.length; ++i) {
+                str += AnsiParser.stringify(arr[i]) + "\n";
+            }
+            str = str.replace(/\n$/, "");
+            return str;
+        }
+
+        var lastStyle = ""
```

Comment:
Why did you change style conventions to use the leading comma?

Reply:
This module was used from a third-party library. It has been moved to `lib` directory.